### PR TITLE
CCP-589: added getTotalProductCount pipeline

### PR DESCRIPTION
**Pull request by aaron-martin**

### CODE REVIEW

File: pipelines/shopgate.catalog.getTotalProductCount.v1.json [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        "input": [
+          {"key": "categoryId", "id": "1", "optional": true},
+          {"key": "searchPhrase", "id": "2", "optional": true},
+          {"key": "filters", "id": "3", "optional": true}
+        ],
+        "output": [
+          {"key": "isSearch", "id": "2001"},
+          {"key": "isSearchFilter", "id": "2002"},
+          {"key": "isCategoryBrowsing", "id": "2003"},
+          {"key": "isProductsById", "id": "2004"}
```
Comment (sofib) [inline pipelines/shopgate.catalog.getTotalProductCount.v1.json]: you don't need this key

File: pipelines/shopgate.catalog.getTotalProductCount.v1.json [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        ]
+      },
+      {
+        "type": "conditional",
+        "input": [
+          {"key": "isCategoryBrowsing", "id": "2003"},
+          {"key": "isProductsById", "id": "2004"}
```
Comment (sofib) [inline pipelines/shopgate.catalog.getTotalProductCount.v1.json]: you don't need this key

File: pipelines/shopgate.catalog.getTotalProductCount.v1.json [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        ],
+        "expression": {
+          "any": [
+            {"ok": [{"type": "input", "name": "isCategoryBrowsing"}]},
+            {"ok": [{"type": "input", "name": "isProductsById"}]}
```
Comment (sofib) [inline pipelines/shopgate.catalog.getTotalProductCount.v1.json]: you don't need this key, ca be a simple expression

File: pipelines/shopgate.catalog.getTotalProductCount.v1.json (hunk 1 above)
Comment (aaron-martin) [reply pipelines/shopgate.catalog.getTotalProductCount.v1.json]: key removed

File: pipelines/shopgate.catalog.getTotalProductCount.v1.json (hunk 2 above)
Comment (aaron-martin) [reply pipelines/shopgate.catalog.getTotalProductCount.v1.json]: key removed.

File: pipelines/shopgate.catalog.getTotalProductCount.v1.json (hunk 3 above)
Comment (aaron-martin) [reply pipelines/shopgate.catalog.getTotalProductCount.v1.json]: expression reduced to simple expression.

**Pull request merged**